WordPress/wordcamp.org
# Blocks: Add query loop variations for WordCamp post types

This adds special query loop variations for the rest of the WordCamp post types (Organizers, Speakers, Sponsors), so that it's easier to use the query loop with our content types.

The Query approach is more flexible, allowing organizers to move around blocks and adjust the formatting on a per-block level. Eventually, these could replace the old list blocks in the new site content. Until then, they're kind of a hidden feature, discovered by searching the blocks list for "speakers", etc. (once this is merged, I'll add another issue to update the new content)

When added, these new blocks set up a default query (10 items per page, sorted A-Z), which can be updated to filter a specific taxonomy (or any other changes). It also drops in set of default blocks, different for each post type.

**Organizers**
- A row with organizer avatar & name
- Post excerpt
- A row with "Teams:" and the teams tagged to this organizer

**Speakers**
- A row with speaker avatar & name
- Post excerpt
- Speaker sessions list, showing session details and linking to each session

**Sponsors**
- Featured image, linked to sponsor post
- Post title, linked to sponsor post
- Post excerpt

This also updates the session list variation with a new icon, and updates the attributes on the group block.

Fixes #797, See #754

### Screenshots

| | Screenshot |
|---|---|
| organizers | ![](https://github.com/WordPress/wordcamp.org/assets/541093/5968adf0-3e5b-493d-9241-258e99d08adc) |
| sessions | ![](https://github.com/WordPress/wordcamp.org/assets/541093/107bac70-c4a7-4b46-8b3c-efb73d29c049) |
| sponsors | ![](https://github.com/WordPress/wordcamp.org/assets/541093/217511f0-337e-48ea-a432-25bea85d9104) |
| speakers | ![](https://github.com/WordPress/wordcamp.org/assets/541093/1f83bf7a-470f-42be-97c3-7177ce152b2a) |

These blocks all have custom icons to differentiate them from the old list blocks:

<img alt="" width="331" src="https://github.com/WordPress/wordcamp.org/assets/541093/ef40c93b-502a-4c8f-816b-d5817faf9fb8" />

### How to test the changes in this Pull Request:

1. Have content set up on your site
2. Add any of these list blocks
3. It should display the content you chose
4. Changing the filters should apply to the content
